test/e2e/specs/site-editor/style-variations.spec.js

```diff
@@ -0,0 +1,266 @@
+			)
+		).toHaveCSS( 'background', /rgb\(255, 239, 11\)/ );
+
+		await expect(
+			page.locator(
+				'css=[aria-label="Colors text styles"] .component-color-indicator'
+			)
+		).toHaveCSS( 'background', /rgb\(25, 25, 17\)/ );
+
+		await siteEditorStyleVariations.openPreviousGlobalStylesPanel();
+		await siteEditorStyleVariations.openTypographyPanel();
+		await siteEditorStyleVariations.openTextPanel();
+
+		expect( await siteEditorStyleVariations.getFontSizeHint() ).toBe(
+			'(Custom)'
+		);
+		expect( await siteEditorStyleVariations.getCustomFontSizeValue() ).toBe(
+			'15'
+		);
+	} );
+
+	test( 'should apply a color palette in a variation', async ( {
+		admin,
+		page,
+		siteEditorStyleVariations,
+	} ) => {
+		await admin.visitSiteEditor( {
+			postId: 'gutenberg-test-themes/style-variations//index',
+			postType: 'wp_template',
+		} );
+		await siteEditorStyleVariations.applyPinkVariation();
+		await siteEditorStyleVariations.openPreviousGlobalStylesPanel();
+		await siteEditorStyleVariations.openColorsPanel();
+		await page.click( `role=button[name="Color palettes"i]` );
+
+		await expect(
+			page.locator( 'role=button[name="Color: Foreground"i]' )
+		).toHaveCSS( 'background', /rgb\(74, 7, 74\)/ );
+
+		await expect(
+			page.locator( 'role=button[name="Color: Background"i]' )
+		).toHaveCSS( 'background', /rgb\(202, 105, 211\)/ );
+
+		await expect(
+			page.locator( 'role=button[name="Color: Awesome pink"i]' )
+		).toHaveCSS( 'background', /rgba\(204, 0, 255, 0\.77\)/ );
+	} );
+
+	test( 'should reflect style variations in the styles applied to the editor', async ( {
+		admin,
+		page,
+		siteEditorStyleVariations,
+	} ) => {
+		await admin.visitSiteEditor( {
+			postId: 'gutenberg-test-themes/style-variations//index',
+			postType: 'wp_template',
+		} );
+		await siteEditorStyleVariations.applyYellowVariation();
+		const frame = page.frameLocator(
+			'css=.edit-site-visual-editor iframe'
+		);
+		const paragraph = frame.locator(
+			'xpath=//p[text()="My awesome paragraph"]'
+		);
+		const paragraphColor = await paragraph.evaluate( ( el ) => {
+			return window.getComputedStyle( el ).color;
+		} );
+		expect( paragraphColor ).toBe( 'rgb(25, 25, 17)' );
+		const body = frame.locator( 'css=body' );
+		const backgroundColor = await body.evaluate( ( el ) => {
+			return window.getComputedStyle( el ).backgroundColor;
+		} );
+		expect( backgroundColor ).toBe( 'rgb(255, 239, 11)' );
+	} );
+} );
+
+class SiteEditorStyleVariations {
+	constructor( { admin, editor, page, pageUtils, requestUtils } ) {
+		this.admin = admin;
+		this.editor = editor;
+		this.page = page;
+		this.pageUtils = pageUtils;
+		this.requestUtils = requestUtils;
+	}
+
+	async openOtherStyles() {
+		await this.page
+			.locator( 'role=button[name="Browse styles"i]' )
+			.waitFor( { state: 'attached' } ); // Wait for the element to appear after the navigation animation.
+
+		await this.page.click( 'role=button[name="Browse styles"i]' );
```

Comment:
Playwright will auto-wait for the button to be clickable. I don't think we need to explicitly wait for it to be attached. This brings up down to this:

```suggestion
		await this.page.click( 'role=button[name="Browse styles"i]' );
```

Which I think we can just inline it without having to create an util here?

Reply:
Yeah I originally tried that because I was experiencing a lot of timeouts, which I assumed were due to the animation.

`waiting for selector "role=button[name="Browse styles"i]"`

But the one liner seems to work locally for me 🤞 

Thanks!

Reply:
Do you remember why it was timing out before? Might want to keep an eye on that in case we accidentally ship a flaky test 😅.

Reply:
> Do you remember why it was timing out before?

No. I was watching the reply in the browser and it I could see the element on the page, but playwright couldn't 🤣 

I only assumed it had something to do with the animation delay 🤷 

I just run the tests 10 x locally with the update and they pass. Let's see....

Reply:
Oh, and now the 11th time:

```bash
    page.click: Timeout 10000ms exceeded.
    =========================== logs ===========================
    waiting for selector "role=button[name="Browse styles"i]"
    ============================================================

      221 | 	async applyVariation( label ) {
      222 | 		await this.page.click( 'role=button[name="Styles"i]' );
    > 223 | 		await this.page.click( 'role=button[name="Browse styles"i]' );
          | 		                ^
      224 | 		await this.page.click( `role=button[name="${ label }"i]` );
      225 | 	}
      226 |
``` 

😮‍💨

Reply:
Ah, the Browse Styles button appears after a slight delay 😄 

![2022-06-17 11 41 33](https://user-images.githubusercontent.com/6458278/174205628-67e46aa8-142a-4b7e-99d2-9c521fecb6a7.gif)

Reply:
Ooooooooooooooooooh 🤦 🤦 🤦 🤦 🤦 🤦 🤦 🤦 🤦 

<img width="641" alt="Screen Shot 2022-06-17 at 12 05 21 pm" src="https://user-images.githubusercontent.com/6458278/174209842-aa807b0d-d7ed-49ca-8dfc-bcb1348053db.png">